openstack/apigw/dedicated/v2/appauths/results.go

```diff
@@ -0,0 +1,168 @@
+package appauths
+
+import "github.com/chnsz/golangsdk/pagination"
+
+type createResp struct {
+	// App authorization records.
+	Auths []Authorization `json:"auths"`
+}
+
+// Authorization is the struct that represents the application authorization records.
+type Authorization struct {
+	// The application ID.
+	ApiId string `json:"api_id"`
+	// The authorization result.
+	AuthResult AuthResult `json:"auth_result"`
+	// The authorization time.
+	AuthTime string `json:"auth_time"`
+	// The authorization record ID.
+	ID string `json:"id"`
+	// The ID of the application allowed to access the API.
+	AppId string `json:"app_id"`
+	// The authorizer.
+	// + PROVIDER: API provider
+	// + CONSUMER: API user
+	AuthRole string `json:"auth_role"`
+	// Authorization channel type.
+	// + NORMAL: normal channel
+	// + GREEN: green channel
+	// The default value is NORMAL.
+	AuthTunnel string `json:"auth_tunnel"`
+	// Whitelist for the green channel.
+	AuthWhitelist []string `json:"auth_whitelist"`
+	// Blacklist for the green channel.
+	AuthBlacklist []string `json:"auth_blacklist"`
+	// Access parameters.
+	VisitParams string `json:"visit_params"`
+}
+
+// AuthResult is the structure that represents the authorization details.
+type AuthResult struct {
+	// Authorization result.
+	// + SUCCESS
+	// + SKIPPED
+	// + FAILED
+	Status string `json:"status"`
+	// Error message.
+	ErrorMsg string `json:"error_msg"`
+	// Error code.
+	ErrorCode string `json:"error_code"`
+	// Name of the API for which authorization fails.
+	ApiName string `json:"api_name"`
+	// Name of the app that fails to be authorized.
+	AppName string `json:"app_name"`
+}
+
+// AuthorizedPage is a page structure that represents each page information.
+type AuthorizedPage struct {
+	pagination.OffsetPageBase
+}
+
+// IsEmpty checks whether a AuthorizedPage struct is empty.
+func (b AuthorizedPage) IsEmpty() (bool, error) {
+	arr, err := ExtractAuthorizedApis(b)
+	return len(arr) == 0, err
+}
+
+// ExtractAuthorizedApis is a method to extract the list of authorized APIs for specified application.
+func ExtractAuthorizedApis(r pagination.Page) ([]ApiAuthInfo, error) {
+	var s []ApiAuthInfo
+	err := r.(AuthorizedPage).Result.ExtractIntoSlicePtr(&s, "auths")
+	return s, err
+}
+
+// ApiAuthInfo is the structure that represents the authorized API information.
+type ApiAuthInfo struct {
```

Comment:
How about rename `ApiAuthInfo` to `ApiBindedInfo`, and rename `ApiOutlineInfo` to `ApiUnBindedInfo`